src/MongoDB.Driver.Core/Core/Compression/Zstandard/PinnedBufferWalker.cs

```diff
@@ -0,0 +1,71 @@
+﻿/* Copyright 2020–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Runtime.InteropServices;
+using MongoDB.Driver.Core.Misc;
+
+namespace MongoDB.Driver.Core.Compression.Zstandard
+{
+    internal class PinnedBufferWalker : IDisposable
+    {
+        private readonly byte[] _bytes;
+        private GCHandle _handle;
```

Comment:
we already discussed it :) Since it's a struct, it's not possible to add readonly here

Reply:
I don't believe that is right.

I am quite certain it can and should be readonly.

Reply:
I had forgotten about the discussion last year about this on the Snappy review. But you are right, if a struct is marked readonly the C# compiler makes a temporary copy of the struct before calling methods like `Free`. 

Because this is such a weird an unexpected quirk of the C# language/compiler let's add a comment to explain why we didn't make it readonly:

    private GCHandle _handle; // not readonly to prevent a temporary copy from being created when calling Free

or some wording to that effect.